File: Qatime_Student/QuestionInfoView.m
//
//  QuestionInfoView.m
//  Qatime_Student
//
//  Created by Shin on 2017/9/27.
//  Copyright © 2017年 WWTD. All rights reserved.
//

#import "QuestionInfoView.h"
#import "NSString+TimeStamp.h"

@implementation QuestionInfoView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        
        _subTitle = [[QuestionSubTitle alloc]initWithFrame:CGRectMake(0, 0, self.width_sd, 100)];
        [self addSubview:_subTitle];
        
        _subTitle.sd_layout
        .topSpaceToView(self, 0)
        .leftSpaceToView(self, 0)
        .rightSpaceToView(self, 0)
        .heightIs(100);
        
        _questionContent = [[UILabel alloc]init];
        [self addSubview: _questionContent];
        _questionContent.font = TEXT_FONTSIZE_MIN;
        _questionContent.textColor = [UIColor blackColor];
        _questionContent.sd_layout
        .topSpaceToView(_subTitle, 20*ScrenScale)
        .leftSpaceToView(self, 10*ScrenScale)
        .rightSpaceToView(self, 10*ScrenScale)
        .autoHeightRatio(0);
        
        UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc]init];
       
        _photosView = [[UICollectionView alloc]initWithFrame:CGRectZero collectionViewLayout:layout];
        _photosView.backgroundColor = [UIColor whiteColor];
        [self addSubview:_photosView];
        _photosView.sd_layout
        .topSpaceToView(_questionContent, 10*ScrenScale)
        .leftSpaceToView(self, 10*ScrenScale)
        .rightSpaceToView(self, 10*ScrenScale)
        .heightIs((UIScreenWidth-40*ScrenScale-10)/5.f);
        
        _recorder = [[YZReorder alloc]init];
        [self addSubview:_recorder.view];
        _recorder.view.sd_layout
        .leftSpaceToView(self, 10*ScrenScale)
        .rightSpaceToView(self, 10*ScrenScale)
        .topSpaceToView(_photosView, 10*ScrenScale)
        .heightIs(40);
        
        //目前先隐藏 , 根据问题实际内容修改.
//        _photosView.hidden = YES;
//        _recorder.view.hidden = YES;
        
//        [self setupAutoHeightWithBottomView:_questionContent bottomMargin:20*ScrenScale];
        
    }
    return self;
}


- (void)setModel:(Questions *)model{
    _model = model;
    _subTitle.title.text = model.title;
    _subTitle.creat_at.text = [NSString stringWithFormat:@"创建时间: %@",[model.created_at changeTimeStampToDateString]];
    _subTitle.name.text = model.user_name;
    _questionContent.text = model.body;
    
    //先判断问题吧.
    if (model.body) {
        model.haveBody = YES;
    }else{
        model.haveBody = NO;
    }
    
    if (model.attachments) {
        for (NSDictionary *atts in model.attachments) {
            if ([atts[@"file_type"]isEqualToString:@"png"]||[atts[@"file_type"]isEqualToString:@"jpg"]||[atts[@"file_type"]isEqualToString:@"jpeg"]) {
                model.havePhotos = YES;
            }else if ([atts[@"file_type"]isEqualToString:@"mp3"]){
                model.haveRecord = YES;
            }
        }
        
    }else{
        model.havePhotos = NO;
        model.haveRecord = NO;
    }
    
    //问题部分.整自动布局....
    if (model.haveBody) {
        _questionContent.hidden = NO;
        if (model.havePhotos) {
            _photosView.hidden = NO;
            if (model.haveRecord) {
                _recorder.view.hidden = NO;
                [self setupAutoHeightWithBottomView:_recorder.view bottomMargin:10];
            }else{
                _recorder.view.hidden = YES;
                [self setupAutoHeightWithBottomView:_photosView bottomMargin:10];
            }
        }else{
            _photosView.hidden = YES;
            if (model.haveRecord) {
                _recorder.view.hidden = NO;
                _recorder.view.sd_layout
                .topSpaceToView(_questionContent, 15);
                [_recorder.view updateLayout];
                [self setupAutoHeightWithBottomView:_recorder.view bottomMargin:10];
            }else{
                _recorder.view.hidden = YES;
                [self setupAutoHeightWithBottomView:_questionContent bottomMargin:20];
            }
            
        }
        
    }else{
        _questionContent.hidden = YES;
        _photosView.sd_layout
        .topSpaceToView(_subTitle, 20*ScrenScale);
        [_photosView updateLayout];
        if (model.havePhotos) {
            _photosView.hidden = NO;
            if (model.haveRecord) {
                _recorder.view.hidden = NO;
                [self setupAutoHeightWithBottomView:_recorder.view bottomMargin:10];
            }else{
                _recorder.view.hidden = YES;
                [self setupAutoHeightWithBottomView:_photosView bottomMargin:10];
            }
        }else{
            _photosView.hidden = YES;
            if (model.haveRecord) {
                _recorder.view.hidden = NO;
                _recorder.view.sd_layout
                .topSpaceToView(_questionContent, 15);
                [_recorder.view updateLayout];
                [self setupAutoHeightWithBottomView:_recorder.view bottomMargin:10];
            }else{
                _recorder.view.hidden = YES;
                [self setupAutoHeightWithBottomView:_questionContent bottomMargin:20];
            }
        }
    }
    
}



@end

File: Qatime_Student/QuestionInfoView.swift
//
//  QuestionInfoView.swift
//  Qatime_Student
//
//  Created by Shin on 2017/8/18.
//  Copyright © 2017年 WWTD. All rights reserved.
//

import UIKit

let ScreenWidth = UIScreen.main.bounds.size.width

class QuestionInfoView: UIView {
    ///副标题
    var subTitle  = QuestionSubTitle()
    
    ///内容
    var questionContent = UILabel()
    
    ///图片
//    var photosView = UICollectionView()
    
    //音频
//    var recorder = YZPlayerRecorder()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        //副标题
        subTitle = QuestionSubTitle.init(frame: CGRect(x:0,y:0,width:self.width_sd,height:100))
        self.addSubview(subTitle)
        subTitle.sd_layout()
        .topSpaceToView(self,0)?
        .leftSpaceToView(self,0)?
        .rightSpaceToView(self,0)?
        .heightIs(100)
        
        //内容
        questionContent = UILabel.init()
        self.addSubview(questionContent)
        questionContent.font = TEXT_FONTSIZE_MIN
        questionContent.textColor = UIColor.black
        questionContent.sd_layout()
        .topSpaceToView(subTitle,20*ScrenScale)?
        .leftSpaceToView(self,10*ScrenScale)?
        .rightSpaceToView(self,10*ScrenScale)?
        .autoHeightRatio(0)
        
//        //图片
//        let layout = UICollectionViewFlowLayout.init()
//        layout.itemSize = CGSize(width:(ScreenWidth-20*ScrenScale-10)*0.5,height:(ScreenWidth-20*ScrenScale-10)*0.5)
//        layout.minimumInteritemSpacing = 10
//        layout.minimumLineSpacing = 10
//        photosView = UICollectionView.init(frame: CGRect(x:0,y:0,width:0,height:0), collectionViewLayout: layout)
//        self.addSubview(photosView)
//        photosView.sd_layout()
//        .topSpaceToView(questionContent,10*ScrenScale)?
//        .leftSpaceToView(self,10*ScrenScale)?
//        .rightSpaceToView(self,10*ScrenScale)?
//        .heightIs((ScreenWidth-20*ScrenScale-10)*0.5*3+20)
//    
//        //录音机
//        recorder = YZPlayerRecorder.init()
//        self.addSubview(recorder.view)
//        recorder.view.sd_layout()
//        .leftSpaceToView(self,10*ScrenScale)?
//        .rightSpaceToView(self,10*ScrenScale)?
//        .topSpaceToView(photosView,10*ScrenScale)?
//        .heightIs(40)
//        
//        ///该隐藏的都隐藏
//        photosView.isHidden = true
//        recorder.view.isHidden = true
        
        self.setupAutoHeight(withBottomView: questionContent, bottomMargin: 20*ScrenScale)
        
    }
    
    func setQuestionMod(model:Questions!) {
        subTitle.title.text = model.title
        subTitle.creat_at.text = "创建时间: " + NSString.changeTimeStamp(toDateString: model.created_at)
        subTitle.name.text = model.user_name
        questionContent.text = model.body
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}
